### PR TITLE
[IOTDB-5201] Extracting getConfig from the registerDataNode interface

### PR DESCRIPTION
**Pull request by Cpaulyz**
## Description

Extracting getConfig from the registerDataNode interface. 

https://apache-iotdb.feishu.cn/docs/doccnMgSBOgHqeT9gjs4RUrqTob

### CODE REVIEW

File: confignode/src/main/java/org/apache/iotdb/confignode/consensus/response/ConfigurationResp.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+public class ConfigurationResp implements DataSet {
+
+  private TSStatus status;
+  private TGlobalConfig globalConfig;
+  private TRatisConfig ratisConfig;
+  private TCQConfig cqConfig;
+
+  public ConfigurationResp() {
+    this.globalConfig = null;
```
Comment (OneSizeFitsQuorum) [inline confignode/src/main/java/org/apache/iotdb/confignode/consensus/response/ConfigurationResp.java:37]: Remove this?

File: confignode/src/main/java/org/apache/iotdb/confignode/consensus/response/DataNodeRegisterResp.java [hunk 2]
```diff
@@ -51,7 +44,6 @@ public class DataNodeRegisterResp implements DataSet {
 
   public DataNodeRegisterResp() {
     this.dataNodeId = null;
```
Comment (OneSizeFitsQuorum) [inline confignode/src/main/java/org/apache/iotdb/confignode/consensus/response/DataNodeRegisterResp.java:46]: Remove this?

**Review by OneSizeFitsQuorum — APPROVED** [review]
LGTM

File: confignode/src/main/java/org/apache/iotdb/confignode/manager/node/NodeManager.java [hunk 3]
```diff
@@ -261,6 +263,18 @@ public DataSet registerDataNode(RegisterDataNodePlan registerDataNodePlan) {
     dataSet.setDataNodeId(
         registerDataNodePlan.getDataNodeConfiguration().getLocation().getDataNodeId());
     dataSet.setConfigNodeList(getRegisteredConfigNodes());
+    return dataSet;
+  }
+
+  /**
+   * Get configuration
+   *
+   * @return DataNodeConfigurationDataSet. The TSStatus will be set to SUCCESS_STATUS when register
```
Comment (Beyyes) [inline confignode/src/main/java/org/apache/iotdb/confignode/manager/node/NodeManager.java:272]: `The TSStatus will be set to SUCCESS_STATUS when register success, and DATANODE_ALREADY_REGISTERED when the DataNode is already exist.`  
This comment is described for method `registerDataNode`, method `getConfiguration` doesn't need it.

**Review by CRZbulabula — APPROVED** [review]
LGTM~

**Review by Beyyes — APPROVED** [review]
LGTM

**Pull request merged**